shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -2167,7 +2249,10 @@ - (void)setSelectionRects:(NSArray*)rects {
         addObject:[FlutterTextSelectionRect
                       selectionRectWithRect:CGRectMake([rect[0] floatValue], [rect[1] floatValue],
                                                        [rect[2] floatValue], [rect[3] floatValue])
-                                   position:[rect[4] unsignedIntegerValue]]];
+                                   position:[rect[4] unsignedIntegerValue]
+                           writingDirection:[rect[5] unsignedIntegerValue] == 1
+                                                ? NSWritingDirectionLeftToRight
+                                                : NSWritingDirectionRightToLeft]];
```

Comment:
im a bit confused here. What does `rects` stand for? is it for a custom encoding?

Reply:
Yes, it is custom encoding in `_PlatformTextInputControl.setSelectionRects` in the framework.

Reply:
ic. i was confused because of the `rect` variable name, which sounds like a CGRect. Maybe call it `rectData` instead?